websites/F/Flat/presence.ts

```diff
@@ -0,0 +1,84 @@
+const presence = new Presence({
+	clientId: "1021848171946704906",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "flat",
+		},
+		{ pathname } = location;
+	if (pathname === "/") presenceData.details = "Viewing the homepage";
+	if (pathname === "/my-library")
+		presenceData.details = "Viewing their library";
+	if (pathname.startsWith("/collection/")) {
+		presenceData.details = "Viewing a collection:";
+		presenceData.state =
+			document.querySelector(".collection-title")?.textContent;
+	}
+	if (pathname === "/shared-with-me")
+		presenceData.details = "Viewing shared scores";
+	if (pathname === "/trash")
+		presenceData.details = "Viewing recently deleted scores";
+	if (pathname.startsWith("/help/")) {
+		if (pathname.includes("/support"))
+			presenceData.details = "Contacting support";
+		else {
+			const title = document.querySelector(".page-title > h1")?.textContent;
+			presenceData.details = title ? "Viewing help:" : "Viewing help";
+			if (title)
+				presenceData.state = title.replace(/^((Flat|Platform) -|Help:) /, "");
+			if (presenceData.state === "Online Music Notation Software")
+				presenceData.state = "About Flat";
+		}
+	}
+	if (pathname.startsWith("/settings"))
+		presenceData.details = "Viewing settings";
+	if (pathname.startsWith("/updates")) presenceData.details = "Viewing updates";
+	if (pathname === "/settings/upgrade")
+		presenceData.details = "Viewing upgrade options";
+	if (pathname.startsWith("/community")) {
+		if (pathname.match(/^\/community\/popular(\/.+)?\/?$/)) {
+			presenceData.details = "Viewing community scores";
+			presenceData.state = `Since: ${
+				document.querySelector('[data-cy="community-date-filter-btn"]')
+					?.textContent
+			}`;
+		}
+		if (pathname.includes("/challenges/"))
+			presenceData.details = "Viewing a challenge";
+	}
+	if (pathname.startsWith("/score/")) {
+		if (pathname.includes("/edit")) {
+			presenceData.details = "Editing a score:";
+			presenceData.state = document.querySelector(".title")?.textContent;
+		} else {
+			presenceData.details = "Viewing a score:";
+			presenceData.state =
+				document.querySelector(".sv-title > h1")?.textContent;
+			presenceData.smallImageKey = (
+				document.querySelector(".user-pp img") as HTMLImageElement
+			).src;
+			presenceData.smallImageText = document.querySelector(".by")?.textContent;
+			if (document.querySelector(".sc-pause"))
+				presenceData.details = "Listening to:";
+		}
+	}
+	if (pathname.startsWith("/@")) {
+		const name = document.querySelector(
+			'[data-cy="printable-name"]'
+		)?.textContent;
+		presenceData.details = "Viewing a user:";
+		presenceData.state = name
+			? `${name} (@${pathname.split("/")[1].slice(1)})`
+			: "Unknown user";
+		presenceData.smallImageKey = (
+			document.querySelector('[data-cy="user-info"] img') as HTMLImageElement
```

Comment:
```suggestion
			document.querySelector<HTMLImageElement>('[data-cy="user-info"] img')
```

Reply:
this fails lint

Reply:
yes, just run the linter again, and it should auto-fix it. It's caused by the extra parenthesis

Reply:
nope
`document.querySelector<HTMLImageElement>('img[data-cy="user-info"]')
`

Reply:
> > nope it should be:
> > `document.querySelector<HTMLImageElement>('img[data-cy="user-info"]')`
> 
> No it shouldn't; the suggested change is correct

Idk i always use the one I suggested but both could work.

Reply:
> > > nope it should be:
> > > `document.querySelector<HTMLImageElement>('img[data-cy="user-info"]')`
> > 
> > 
> > No it shouldn't; the suggested change is correct
> 
> Idk i always use the one I suggested but both could work.

There's a major difference between the two selectors. Looking at the website, the image itself doesn't have the `data-cy=user-info` property, while its ancestor container does, so the originally suggested change is correct, as said by OP.

Reply:
Noted. Thanks.